fanduelOdds.php

```diff
@@ -35,13 +94,218 @@ private function getLiveMarketIds() {
+            curl_close($liveMarketsCh);
+            return [];
+        } 
+        else {
+            $data = json_decode($liveMarketsResponse);
+            curl_close($liveMarketsCh);
+
+            $liveMarketIds = [];
+            // grab all liveMarketIds
+            foreach ($data as $item) {
+                array_push($liveMarketIds, $item->marketId);
+            }
+            return $liveMarketIds;
+        }
+    }
+    private function getMarketPrices($liveMarketIds) {
         /**
          * Helper function that returns market prices for Spread Betting, Money Line and Total Points for next Raptors game
          *
          * @return array Array of market prices for next Raptors game
          */
+        
+        // setup request to ListMarketPrices API
+        $marketPricesRequestPayload = json_encode([
+            "listMarketPricesRequestParams" => [
+                "marketIds" => $liveMarketIds,
+            ]
+        ]);
+
+        $requestHeaders = [
+            "Content-type: application/json",
+            "X-Application: " . $this->secret_key
+        ];
+    
+        $marketPricesCh = curl_init();
+    
+        curl_setopt_array($marketPricesCh, [
+            CURLOPT_URL => "https://affiliates.sportsbook.fanduel.com/betting/rest/v1/listMarketPrices/",
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_CUSTOMREQUEST => "POST",
+            CURLOPT_POSTFIELDS => $marketPricesRequestPayload,
+            CURLOPT_HTTPHEADER => $requestHeaders
+        ]);
+    
+        $marketPricesResponse = curl_exec($marketPricesCh);
+    
+        if (curl_errno($marketPricesCh)) {
+            echo 'Error (getMarketPrices): ' . curr_error($marketPricesCh);
+            curl_close($marketPricesCh);
+            return [];
+        } 
+        else {
+            $data = json_decode($marketPricesResponse);
+            curl_close($marketPricesCh);
+            
+            $groupedMarketsByGame = new stdClass();
+            $raptorsEventId = "";
+            //group market prices by eventId (id of the game)
+            foreach($data->marketDetails as $market_detail) {
+                $eventId = $market_detail->eventId;
+    
+                // Initialize the array for this eventId if it hasn't been created yet
+                if (!isset($groupedMarketsByGame->$eventId)) {
+                    $groupedMarketsByGame->$eventId = [];
+                }
+                array_push($groupedMarketsByGame->$eventId, $market_detail);
+    
+                foreach($market_detail->runnerDetails as $runnerDetail) {
+                    if ($runnerDetail->selectionId == self::RAPTORS_ID) {
+                        // grab the eventId of the Raptors game
+                        $raptorsEventId = $eventId;
+                    }
+                }
+            }
+            if (!isset($groupedMarketsByGame->$raptorsEventId)) {
+                return [];
```

Comment:
This is returning an array here, but a string two lines later? What does this function return?

Reply:
It's still returning an array two lines later. It's returning all the market prices for the specific Raptors game